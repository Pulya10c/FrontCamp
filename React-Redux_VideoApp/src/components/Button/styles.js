import styled from 'styled-components';

export const Button = styled.button`
  font-family: 'Open Sans', sans-serif;
  font-weight: 800;
  font-size: 18px;
  line-height: 24px;
  outline: none;
  background-color: #004db9;
  border: 0;
  border-radius: 4px;
  color: white;
  padding: 15px;
  cursor: pointer;
  transition: all 0.3s cubic-bezier(0.075, 0.82, 0.165, 1);

  &:disabled {
    cursor: not-allowed;
    background-color: #7792b8;
  }

  &:hover:not(:disabled) {
    background-color: #00183a;
  }

  &:active {
    background-color: #004db9;
    opacity: 0.7;
  }
`;
