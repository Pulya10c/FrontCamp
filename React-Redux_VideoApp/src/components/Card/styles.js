import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const CardStyled = styled(Link)`
  display: grid;
  grid-gap: 12px;
  align-content: start;
  padding: 4px;
  font-family: 'Open Sans', sans-serif;
  border-radius: 10px;
  background-color: #fff;
  border: 1px solid rgba(218, 220, 224, 0.5);
  box-shadow: 0 20px 20px -20px $main-color;
  transition: all 0.3s cubic-bezier(0.075, 0.82, 0.165, 1);
  cursor: pointer;
  &:hover {
    transform: scale(1.02);
    box-shadow: 0 2px 2px #00183a;
  }
`;
