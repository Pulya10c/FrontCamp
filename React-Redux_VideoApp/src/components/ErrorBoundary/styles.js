import styled, { css } from 'styled-components'

export const commonAnimation = css`
  opacity: 0;
  animation: appear 500ms cubic-bezier(0, 1, 0.5, 1) 1 normal forwards;

  @keyframes appear {
    0% {
      opacity: 0;
      transform: translateY(100%);
    }
    100% {
      opacity: 1;
      transform: translateY(0);
    }
  }
`

export const ErrorWrapper = styled.div`
  margin: 40px auto;
  text-align: center;
`

export const ErrorTitle = styled.h1`
  ${commonAnimation}

  font-size: 40px;
  animation-delay: 0;
`

export const ErrorSubTitle = styled.h2`
  ${commonAnimation}

  font-size: 30px;
  animation-delay: 200ms;
`

export const ErrorContent = styled.p`
  ${commonAnimation}

  font-size: 18px;
  margin: 2em;
  animation-delay: 500ms;
`
