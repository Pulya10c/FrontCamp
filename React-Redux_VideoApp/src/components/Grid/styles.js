import styled from 'styled-components';
import { mediaMd, mediaSm } from 'utils/style-utils';

export const Grid = styled.section`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 5px;
  font-size: 20px;
  padding: 5px;

  ${mediaMd`
    grid-template-columns: repeat(3, 1fr);
    font-size: 24px;
  `};

  ${mediaSm`
    grid-template-columns: repeat(1, 1fr);
    font-size: 20px;
  `};
`;
