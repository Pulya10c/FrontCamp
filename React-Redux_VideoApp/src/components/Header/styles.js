import styled from 'styled-components';
import blurImage from './blurImage.jpg';

export const Header = styled.header`
  display: grid;
  grid-gap: 10px;
  justify-content: center;
  align-content: center;

  padding: 10px;
  min-height: 200px;
  background-image: url(${blurImage});
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
`;
