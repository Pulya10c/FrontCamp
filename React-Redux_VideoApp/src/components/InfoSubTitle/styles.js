import styled from 'styled-components';
import { mediaMd, mediaSm } from 'utils/style-utils';

export const InfoSubTitle = styled.h5`
  font-size: 16px;
  color: white;

  ${mediaMd`
    font-size: 14px;
  `};

  ${mediaSm`
    font-size: 12px;
  `};
`;
