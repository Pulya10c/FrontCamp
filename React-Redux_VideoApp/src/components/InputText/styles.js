import styled from 'styled-components';

export const SearchInputText = styled.input`
  padding: 5px 10px;
  font-family: 'Open Sans', sans-serif;
  font-weight: 400;
  font-size: 16px;
  line-height: 20px;
  background: white;
  border: 1px solid rgba(31, 64, 111, 0.2);
  border-radius: 4px;

  &:focus {
    border-color: #004db9;
  }
`;
