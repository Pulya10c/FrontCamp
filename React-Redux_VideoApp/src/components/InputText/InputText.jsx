import React from "react";
import PropTypes from "prop-types";

import { SearchInputText } from "./styles";

const InputText = ({ inputValue, executeActionByEnter, setInputValue }) => {
  const handleSearchKeyUp = ({ key }) => {
    if (key === "Enter") {
      executeActionByEnter();
    }
  };

  const handleSearchChange = ({ target: { value } }) => {
    setInputValue(value);
  };

  return (
    <SearchInputText
      type="text"
      value={inputValue}
      onChange={handleSearchChange}
      onKeyUp={handleSearchKeyUp}
    />
  );
};

InputText.propTypes = {
  inputValue: PropTypes.string,
  executeActionByEnter: PropTypes.func,
  setInputValue: PropTypes.func
};

export default InputText;
