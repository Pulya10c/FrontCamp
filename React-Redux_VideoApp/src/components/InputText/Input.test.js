import React from 'react';
import { shallow } from 'enzyme';
import InputText from './InputText';
import { SearchInputText } from './styles';

describe('Component InputText', () => {
  let wrapper, defaultProps;

  beforeEach(() => {
    defaultProps = {
      inputValue: 'mockInputValue',
      setInputValue: jest.fn(),
      executeActionByEnter: jest.fn()
    };
    wrapper = shallow(<InputText {...defaultProps} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call defaultProps.setInputValue with correct args if call onChange', () => {
   const mockVal = 'mockVal'

    wrapper
      .find(SearchInputText)
      .props()
      .onChange({ target: { value: mockVal } });
    expect(defaultProps.setInputValue).toHaveBeenCalledWith(mockVal);
  });

  it('should call defaultProps.executeActionByEnter if call onKeyUp and key === ENTER', () => {
    wrapper
      .find(SearchInputText)
      .props()
      .onKeyUp({ key: 'Enter' });
    expect(defaultProps.executeActionByEnter).toHaveBeenCalledTimes(1);
  });

  it('should not call defaultProps.executeActionByEnter if call onKeyUp and key !== ENTER', () => {
    wrapper
      .find(SearchInputText)
      .props()
      .onKeyUp({ key: 'Space' });
    expect(defaultProps.executeActionByEnter).toHaveBeenCalledTimes(0);
  });
});
