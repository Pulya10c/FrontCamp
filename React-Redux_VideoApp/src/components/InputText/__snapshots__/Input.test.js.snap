// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InputText should render correctly 1`] = `
<styled.input
  onChange={[Function]}
  onKeyUp={[Function]}
  type="text"
  value="mockInputValue"
/>
`;
